coordinator/ttime/ttime_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Edgeless Systems GmbH.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package ttime
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/cloudflare/roughtime/config"
+	"github.com/cloudflare/roughtime/mjd"
+	"github.com/cloudflare/roughtime/protocol"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRoughtime(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+
+	// Create mock server.
+	server, configs, err := NewRoughtimeMockServer("127.0.0.1:2002")
+	require.NoError(err)
+
+	// Start server.
+	server.Start()
+	defer server.Close()
+
+	// Create trusted time client.
+	tt := TrustedTime{servers: configs}
+
+	// Obtain time.
+	rt, err := tt.Roughtime()
+	require.NoError(err)
+
+	// Compare times. Obviously, this is a bidirectional test that heavily
+	// depends on the correctness of testing hardware's local time.
+	// TODO(katexochen): Maybe remove this later.
```

Comment:
Ideally, this would not be part of a Unit test.
Maybe an integration test, but better our e2e-test.